R/qra_fit_convenience.R

```diff
@@ -130,233 +66,132 @@ get_candidate_models <- function(
-
-  results <- purrr::map_dfr(
-    model_abbrs,
-    function(model_abbr) {
-      # identify submission files within specified dates
-      results_paths <- Sys.glob(paste0(submissions_root, model_abbr,
-          "/*", submission_dates, "-", model_abbr, ".csv"))
-
-      if (keep_last) {
-        results_paths <- tail(results_paths, 1)
-      }
-
-      # if no results, return
-      if (length(results_paths) == 0) {
-        return(NULL)
-      }
-
-      purrr::map_dfr(
-        results_paths,
-        function(results_path) {
-          readr::read_csv(results_path,
-            col_types = cols(
-              forecast_date = col_date(format = "%Y-%m-%d"),
-              target = col_character(),
-              target_end_date = col_date(format = "%Y-%m-%d"),
-              location = col_character(),
-              type = col_character(),
-              quantile = col_double(),
-              value = col_double()
-            )) %>%
-            dplyr::filter(
-              tolower(type) %in% types,
-              tolower(target) %in% targets) %>%
-            dplyr::transmute(
-              model = model_abbr,
-              timezero = forecast_date,
-              location = location,
-              target = tolower(target),
-              target_end_date = target_end_date,
-              quantile = quantile,
-              value = value
-            )
-        }
-      )
-    }
-  )
-
-  return(results)
-}
-
-#' Load forecasts from multiple models over multiple weeks, keep only the last
-#' forecast from each model each week, and adjust the targets for daily
-#' forecasts to be relative to the Monday on or after submission.
-#'
+#' Load multiple models' last forecasts from multiple weeks and 
+#' adjust the targets for daily forecasts to be relative to the 
+#' Monday on or after submission.
+#' 
+#' @param hub character vector, where the first element indicates the hub
+#' from which to load forecasts. Possible options are "US" and "ECDC"
+#' @param source string specifying where forecasts will be loaded from: either 
+#' "local_hub_repo" or "zoltar"
+#' @param hub_repo_path path to local clone of the reichlab/covid19-forecast-hub
+#' repository to be used when `source` is `local_hub_repo`
 #' @param monday_dates Date vector of Mondays that are submission deadlines
 #' @param model_abbrs Character vector of model abbreviations
 #' @param timezero_window_size The number of days back to go.  A window size of
 #' 0 will retrieve only forecasts submitted on the `last_timezero` date.
 #' @param locations character vector of locations; FIPS codes
 #' @param targets character vector of targets to retrieve, for example
 #' c('1 wk ahead cum death', '2 wk ahead cum death')
-#' @param horizon maximum horizon relative to the monday dates that should be
-#' retained
+#' @param max_horizon maximum horizon relative to forecast_week_end_date and 
+#' analogous training reference dates for both training and prediction
```

Comment:
Attempting to avoid the forecast_week_end_date terminology
```suggestion
#' @param max_horizon maximum horizon of forecasts to return, where horizon is relative to
#' the reference dates for the forecasts (Saturdays for forecasts of weekly targets or Mondays for
#' daily targets)
```